subsys/bluetooth/host/hci_core.c

```diff
@@ -4802,6 +4859,21 @@ int bt_le_adv_start(const struct bt_le_adv_param *param,
 		return err;
 	}
 
+	if (param->options & BT_LE_ADV_OPT_USE_NAME) {
+		const char *name;
+
+		/* Cannot use name if sd is set */
+		if (sd) {
+			return -EINVAL;
+		}
```

Comment:
Can't we do better than this? I think it would be friendlier if we'd just append the user-given SD elements after the name. I think there may well be valid use cases for other elements in the SD while still wanting to use the configured name there as well.

Reply:
Hmm, like replacing the SD with another array and then making that point to user SD? I guess that could work but what I was thinking is that we would need to check for the name to be part of the SD already in which case we would have to fail.